File: Programmers/[92344] 파괴되지 않은 건물/KingDonggyu/92344.js
function solution(board, skill) {
  const [N, M] = [board.length, board[0].length];
  const degreeSum = Array.from(Array(N + 1), () => Array(M + 1).fill(0));

  skill.forEach(([type, r1, c1, r2, c2, degree]) => {
    if (type === 1) {
      degree *= -1;
    }
    degreeSum[r1][c1] += degree;
    degreeSum[r2 + 1][c1] += -degree;
    degreeSum[r1][c2 + 1] += -degree;
    degreeSum[r2 + 1][c2 + 1] += degree;
  });

  // 행 누적합
  for (let x = 0; x <= N; x++) {
    for (let y = 1; y <= M; y++) {
      degreeSum[x][y] += degreeSum[x][y - 1];
    }
  }

  let answer = 0;

  // 열 누적합
  for (let y = 0; y <= M; y++) {
    for (let x = 1; x <= N; x++) {
      if (degreeSum[x - 1][y] + board[x - 1][y] > 0) {
        answer++;
      }
      degreeSum[x][y] += degreeSum[x - 1][y];
    }
  }

  return answer;
}

File: Programmers/[92344] 파괴되지 않은 건물/Johoseong/92344.py
def solution(board, skill):
    answer = 0
    R = len(board)
    C = len(board[0])
    sum_attack = [[0] * (C + 1) for _ in range(R + 1)]

    # 누적합 연산 위해서 skill에 따라 변화량 세팅
    for t, r1, c1, r2, c2, degree in skill:
        amount = degree
        if t == 1: amount *= (-1) # 공격이면 음수
        
        sum_attack[r1][c1] += amount
        sum_attack[r2 + 1][c2 + 1] += amount
        sum_attack[r1][c2 + 1] += (amount * -1)
        sum_attack[r2 + 1][c1] += (amount * -1)

    for r in range(R): # 누적합 연산 (가로)
        for c in range(C):
            sum_attack[r][c + 1] += sum_attack[r][c]
    for r in range(R): # 누적합 연산 (세로)
        for c in range(C):
            sum_attack[r + 1][c] += sum_attack[r][c]
    for r in range(R):
        for c in range(C):
            if board[r][c] + sum_attack[r][c] > 0:
                answer += 1
    return answer